### PR TITLE
week1 homework Bassam

### PR DESCRIPTION
**Pull request by BassamHajar**
Thanks & regards,,,

### CODE REVIEW

File: Week1/homework/createAndSeed.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+async function seedDatabase() {
+  connection.connect();
+  try {
+    await execQuery(DATABASE_WORLD);
+    await execQuery(COUNTRIES_TABLE);
+    await execQuery(CITIES_TABLE);
+
+    countries.forEach(async country => {
+      await execQuery(`INSERT INTO countries SET ?`, country);
+      if (countries.indexOf(country) === countries.length - 1)
```
Comment (zhaobn) [inline Week1/homework/createAndSeed.js]: Don't `indexOf` always give you 0 or 1? What do you want to compare here?

File: Week1/homework/createAndSeed.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        console.log(`countries have been added...`);
+    });
+
+    cities.forEach(async city => {
+      await execQuery(`INSERT INTO cities SET ?`, city);
+      if (cities.indexOf(city) === cities.length - 1) console.log(`cities have been added...`);
```
Comment (zhaobn) [inline Week1/homework/createAndSeed.js]: Same here

File: Week1/homework/databaseAndTables.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+// DATABASE
+const DATABASE_WORLD = `CREATE DATABASE IF NOT EXISTS world`;
+
+// COUNTRIES TABLE
+const COUNTRIES_TABLE =
+  'CREATE TABLE IF NOT EXISTS countries (country_name VARCHAR(40) NOT NULL, country_code INT NOT NULL, country_population FLOAT NOT NULL, continent VARCHAR(20) NOT NULL, surface_areas FLOAT NOT NULL)';
+
+// CITIES TABLE
+const CITIES_TABLE =
+  'CREATE TABLE IF NOT EXISTS cities (city_name VARCHAR(20) NOT NULL, country_Abbreviation_code VARCHAR(3) NOT NULL, city_population FLOAT NOT NULL)';
```
Comment (zhaobn) [inline Week1/homework/databaseAndTables.js]: Nice to you snake_case for column name, an extra advice is to keep everything lower-case.

File: Week1/homework/queries.js [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+        result.forEach((country, index) => console.log(`${index + 1}- ${country.country_name}`));
+      }
+    );
+    connection.end();
+  },
+
+  // 2. What are the names of countries that have “land” in their names ?
+  two: () => {
+    connection.query(
+      'SELECT DISTINCT country_name FROM countries WHERE country_name LIKE "%land%"',
```
Comment (zhaobn) [inline Week1/homework/queries.js]: Very good to use DISTINCT 👍

File: Week1/homework/queries.js [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+        result.forEach(city => console.log(`\nPopulation of Rotterdam: ${city.city_population}`));
+      }
+    );
+    connection.end();
+  },
+
+  // 8. What's the top 10 countries by Surface Area ?
+  eight: () => {
+    connection.query(
+      'SELECT DISTINCT country_name, surface_areas FROM countries ORDER BY surface_areas DESC',
```
Comment (zhaobn) [inline Week1/homework/queries.js]: LIMIT 10

File: Week1/homework/queries.js [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+        );
+      }
+    );
+    connection.end();
+  },
+
+  // 9. What's the top 10 most populated cities?
+  nine: () => {
+    connection.query(
+      'SELECT DISTINCT city_name, city_population FROM cities ORDER BY city_population DESC',
```
Comment (zhaobn) [inline Week1/homework/queries.js]: LIMIT 10

File: Week1/homework/createAndSeed.js (hunk 1 above)
Comment (BassamHajar) [reply Week1/homework/createAndSeed.js]: Actually I've put this condition in order to console log a text once the last country of the countries array is been inserted.
I didn't intend to have a comparison here.

File: Week1/homework/databaseAndTables.js (hunk 3 above)
Comment (BassamHajar) [reply Week1/homework/databaseAndTables.js]: Thanks for the advice, I'll stick to it.

File: Week1/homework/queries.js (hunk 5 above)
Comment (BassamHajar) [reply Week1/homework/queries.js]: Yeah sure, I'll push the changes asap.

**Pull request closed without merging**